src/parser.js

```diff
@@ -255,7 +255,16 @@ const grammar = {
             ['. SYMBOL ( arguments )', code`method.$2(/*@2*/current, $4)`],
             ['.. SYMBOL', code`fn.recursive(/*@2*/current, "$2")`],
             ['..( block )', code`fn.recursive(current, current => { $2 })`],
-            ['.[ block ]', code`fn.filter(current, current => { $2 })`]
+            ['.[ block ]', code`fn.filter(current, current => { $2 })`],
+            ['[ e : e ]', code`fn.slice(current, $2, $4, 1)`],
```

Comment:
Let's try to collapse grammar, may be like that:
```
            ['[ sliceNotationComponent ]', code`fn.slice(current, 0, $2)`],
            ['[ sliceNotationComponent sliceNotationComponent ]', code`fn.slice(current, 0, $2, $3)`],
            ['[ e sliceNotationComponent ]', code`fn.slice(current, $2, $3)`],
            ['[ e sliceNotationComponent sliceNotationComponent ]', code`fn.slice(current, $2, $3, $4)`]
        ],

        sliceNotationComponent: [
            [':', code`undefined`],
            [': e', code`$2`]
        ],
```

Reply:
One more try
```js
        queryRoot: [
            ...
            ['[ sliceNotation ]', code`fn.slice(current, $2)`]
        ],

        relativePath: [
            ...
            ['query [ sliceNotation ]', code`fn.slice($1, $3)`]
        ],

        sliceNotation: [
            ['sliceNotationComponent', code`0, $1`],
            ['sliceNotationComponent sliceNotationComponent', code`0, $1, $2`],
            ['e sliceNotationComponent', code`$1, $2`],
            ['e sliceNotationComponent sliceNotationComponent', code`$1, $2, $3`]
        ],

        sliceNotationComponent: [
            [':', code`undefined`],
            [': e', code`$2`]
        ],
```
`sliceNotation` can be changed to include `[]` into its syntaxes